File: app/src/store/seed.ts
// 初回起動時のサンプルデータ（デザインリファレンスのモノたち）
import type { Database } from './types';
import { DEFAULT_SETTINGS } from './types';

const t = Date.now();

export function seedDatabase(): Database {
  return {
    version: 1,
    onboarded: false,
    settings: { ...DEFAULT_SETTINGS },
    children: [
      { id: 'sota', name: 'そうた', age: 6, tone: '#E08A63', createdAt: t - 6 },
      { id: 'mio', name: 'みお', age: 3, tone: '#8BA36F', createdAt: t - 5 },
    ],
    items: [
      {
        id: 'bear', childId: 'sota', name: 'くまのプーさん', year: 2019, season: '春',
        memo: 'はじめて自分でえらんだ ぬいぐるみ。毎ばん いっしょに ねていました。',
        frames: [], photos: [], voice: null, tone: '#E08A63',
        pos: { x: -0.32, z: 0.1 }, createdAt: t - 60,
      },
      {
        id: 'clay', childId: 'sota', name: 'ねんど工作', year: 2023, season: '夏',
        memo: 'ようちえんで つくった さくひん。「これパパ！」と教えてくれた。',
        frames: [], photos: [], voice: null, tone: '#8BA36F',
        pos: { x: 0.4, z: 0.25 }, createdAt: t - 40,
      },
      {
        id: 'draw', childId: 'sota', name: 'かぞくの絵', year: 2024, season: '秋',
        memo: '5人かぞくを おおきく描いた絵。すみっこに いぬのポチも。',
        frames: [], photos: [], voice: null, tone: '#C6A05E',
        pos: { x: 0.56, z: -0.45 }, createdAt: t - 20,
      },
      {
        id: 'shoes', childId: 'mio', name: 'はじめての靴', year: 2021, season: '秋',
        memo: 'あるきはじめた日に はいていた 小さな靴。',
        frames: [], photos: [], voice: null, tone: '#E08A63',
        pos: { x: -0.4, z: 0.18 }, createdAt: t - 50,
      },
      {
        id: 'toy', childId: 'mio', name: 'がらがら', year: 2021, season: '冬',
        memo: 'にぎると よろこんで ふっていた おもちゃ。',
        frames: [], photos: [], voice: null, tone: '#8BA36F',
        pos: { x: 0.28, z: 0.05 }, createdAt: t - 45,
      },
      {
        id: 'photo', childId: 'mio', name: 'たんじょう日', year: 2023, season: '春',
        memo: '1さいの おたんじょう日。ケーキに手をのばして。',
        frames: [], photos: [], voice: null, tone: '#C6A05E',
        pos: { x: 0.52, z: -0.4 }, createdAt: t - 30,
      },
    ],
  };
}

File: app_flutter/lib/data/seed.dart
// 初回起動時のサンプルデータ（デザインリファレンスのモノたち）
import '../models/models.dart';

Database seedDatabase() {
  final t = nowMs();

  Child child(String id, String name, int age, String tone, int back) => Child(
    id: id,
    name: name,
    age: age,
    tone: tone,
    createdAt: t - back,
    updatedAt: t - back,
  );

  MemoryItem item({
    required String id,
    required String childId,
    required String name,
    required int year,
    required Season season,
    required String memo,
    required String tone,
    required RoomPos pos,
    required int back,
  }) => MemoryItem(
    id: id,
    childId: childId,
    name: name,
    year: year,
    season: season,
    memo: memo,
    frames: const [],
    photos: const [],
    voice: null,
    pos: pos,
    tone: tone,
    createdAt: t - back,
    updatedAt: t - back,
  );

  return Database(
    onboarded: false,
    settings: const AppSettings(),
    children: [child('sota', 'そうた', 6, '#E08A63', 6), child('mio', 'みお', 3, '#8BA36F', 5)],
    items: [
      item(
        id: 'bear',
        childId: 'sota',
        name: 'くまのプーさん',
        year: 2019,
        season: Season.haru,
        memo: 'はじめて自分でえらんだ ぬいぐるみ。毎ばん いっしょに ねていました。',
        tone: '#E08A63',
        pos: const RoomPos(-0.32, 0.1),
        back: 60,
      ),
      item(
        id: 'clay',
        childId: 'sota',
        name: 'ねんど工作',
        year: 2023,
        season: Season.natsu,
        memo: 'ようちえんで つくった さくひん。「これパパ！」と教えてくれた。',
        tone: '#8BA36F',
        pos: const RoomPos(0.4, 0.25),
        back: 40,
      ),
      item(
        id: 'draw',
        childId: 'sota',
        name: 'かぞくの絵',
        year: 2024,
        season: Season.aki,
        memo: '5人かぞくを おおきく描いた絵。すみっこに いぬのポチも。',
        tone: '#C6A05E',
        pos: const RoomPos(0.56, -0.45),
        back: 20,
      ),
      item(
        id: 'shoes',
        childId: 'mio',
        name: 'はじめての靴',
        year: 2021,
        season: Season.aki,
        memo: 'あるきはじめた日に はいていた 小さな靴。',
        tone: '#E08A63',
        pos: const RoomPos(-0.4, 0.18),
        back: 50,
      ),
      item(
        id: 'toy',
        childId: 'mio',
        name: 'がらがら',
        year: 2021,
        season: Season.fuyu,
        memo: 'にぎると よろこんで ふっていた おもちゃ。',
        tone: '#8BA36F',
        pos: const RoomPos(0.28, 0.05),
        back: 45,
      ),
      item(
        id: 'photo',
        childId: 'mio',
        name: 'たんじょう日',
        year: 2023,
        season: Season.haru,
        memo: '1さいの おたんじょう日。ケーキに手をのばして。',
        tone: '#C6A05E',
        pos: const RoomPos(0.52, -0.4),
        back: 30,
      ),
    ],
  );
}
